SimpleITK/utilities/dask.py

```diff
@@ -0,0 +1,53 @@
+import dask.array as da
+import SimpleITK as sitk
+from pathlib import Path
+
+
+def from_sitk(filename: Path, chunks=None):
+    """
+    Reads the filename into a dask array with map_block.
+
+    SimpleITK is used to read chunks from the file if supported, otherwise the entire image will be read for each
+    chunk request.
+
+    ITK support full streaming includes MHA, MRC, NRRD and NIFTI file formats.
+    """
+    reader = sitk.ImageFileReader()
+    reader.SetFileName(str(filename))
+    reader.ReadImageInformation()
+    img_shape = reader.GetSize()[::-1]
+    # default to loading the whole image from file
+    if chunks is None:
+        chunks = (-1,) * reader.GetDimension()
```

Comment:
Possibly add documentation here? 

If I understand correctly the -1 means no chunking (dask [automatic chunking docs](https://docs.dask.org/en/stable/array-chunks.html#automatic-chunking)). Wouldn't it be better to use 'auto'. As a user, even when I am aware of chunking I don't know how to do it well, other than z-slice chunks. I assume there is a reason to choose no-chunking over `auto` so worth documenting it.

Reply:
The line a couple above says: "default to loading the whole image from file". Defaulting to 'auto' would not be a good options because 1: If the specific ImageIO does not support streaming then the whole image would be read for each chunk extracted 2: "auto" is unaware of the data layout of the image one disk, the z-slice would be a sensible default for ITK ImageIO which support streaming and the pixel data is store just as one block.

Would changing the above to "default to loading the whole image from file in case the ImageIO does not support streaming" work?

Reply:
The user will need to read the Dask documentation.